src/_cffi_src/openssl/rsa.py

```diff
@@ -92,4 +100,80 @@
 static const long Cryptography_HAS_RSA_OAEP_MD = 0;
 int (*EVP_PKEY_CTX_set_rsa_oaep_md)(EVP_PKEY_CTX *, EVP_MD *) = NULL;
 #endif
+
+/* These functions were added in OpenSSL 1.1.0-pre5 (beta2) */
+#if OPENSSL_VERSION_NUMBER < 0x10100005 || defined(LIBRESSL_VERSION_NUMBER)
+int RSA_set0_key(RSA *r, BIGNUM *n, BIGNUM *e, BIGNUM *d)
+{
+    /* d is the private component and may be NULL */
+    if (n == NULL || e == NULL)
```

Comment:
OpenSSL master has a check for `r->n == n`, without that you can get a double-free.